modules/spark/src/main/scala_2.10/spark-last/notebook/kernel/Repl.scala

```diff
@@ -180,10 +183,29 @@ class Repl(val compilerOpts: List[String], val jars:List[String]=Nil) {
         val request:interp.Request = interp.getClass.getMethods.find(_.getName == "prevRequestList").map(_.invoke(interp)).get.asInstanceOf[List[interp.Request]].last
         //val request:Request = interp.prevRequestList.last
 
+        request.handlers.map { h =>
+          val l = h.definesTerm.map(_.encoded)
+          val r = h.definesType.map(_.encoded)
+          val lr = (l,r) match {
+            case (Some(l), _) =>
+              println(interp.symbolOfTerm(l))
+              Some((Left(l),  interp.classOfTerm(l).map(_.getName)))
+            case (_, Some(r)) =>
+              println(interp.symbolOfTerm(r))
+              Some((Right(r), interp.classOfTerm(r).map(_.getName)))
+            case _ => None
+          }
+          reportDefinitions(lr.map(_._1),
+                            lr.flatMap(_._2).getOrElse("<unknown-type>"),
+                            h.referencedNames.toList.map(_.encoded)
+                          )
+        }
```

Comment:
Dunno yet for 2.11 since a lot changes between the two version in spark... like 2.11  uses regular scala repl.
to be checked, and yup tests will have to come along but this is of course bug free ^^ :-$